firmware/application/apps/ui_fileman.cpp

```diff
@@ -267,14 +267,15 @@ void FileManagerView::on_refactor(NavigationView& nav) {
 		std::string destination_path = current_path.string();
 		if (destination_path.back() != '/')
 			destination_path += '/';
-		destination_path = destination_path + buffer;
+
+		destination_path = get_selected_path().string().back() != '/' ? destination_path + buffer + extension_buffer : destination_path + buffer;
```

Comment:
unclear and complicated ternary operation. use a if else like and clarify what you're doing here.

Reply:
ty, I can but from what i saw ``?:`` is same. and what about files getting deleted?

Reply:
Something like this i.e :

if( get_selected_path().string().back() != '/' )  
{
    //adding back extension since it's not a directory
    destination_path += buffer + extension_buffer ;
}
else  
{
    destination_path += buffer ;
}

Reply:
git killed my indentation but you got the idea.

Reply:
updated.. bug still there, after i constantly renaming 5 or 6 files, files getting deleted and the extension_buffer becoming empty no matter what i do.

Reply:
> ty, I can but from what i saw `?:` is same. and what about files getting deleted?
It's less readable and not quicker, maybe only IMHO.
I do not get the file deleted thing, you have to give more details.

Reply:
>I do not get the file deleted thing, you have to give more details.

I'm renaming files, all good, but at some point, the file I'm renaming, getting deleted. it shouldn't happen.

Reply:
>ho you mean the content of the var is dead ?
no.
the real file at the sdcard getting deleted...
Sounds weird but it happened.